src/api/mutations/usePostEventToGoogleCalendar.tsx

```diff
@@ -0,0 +1,36 @@
+import { Session } from '@supabase/supabase-js';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+
+export interface PostEvent {
+  end: { dateTime: string };
+  start: { dateTime: string };
+  summary: string;
+}
+
+const postEvents = async (session: Session | null, event: PostEvent | null) => {
+  const url = new URL('https://www.googleapis.com/calendar/v3/calendars/primary/events');
+
+  if (session) {
+    const res = await fetch(url, {
+      method: 'POST',
+      headers: {
+        Authorization: 'Bearer ' + session.provider_token,
+        'Content-Type': 'application/json',
+      },
```

Comment:
skoro tu zawsze będzie ten Bearer to może warto z axiosa skorzystać i sobie skonfigurować takiego http klienta